connector/connect/common/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -378,10 +379,6 @@ message Sample {
 
   // (Optional) The random seed.
   optional int64 seed = 5;
-
-  // (Optional) Explicitly sort the underlying plan to make the ordering deterministic.
-  // This flag is only used to randomly splits DataFrame with the provided weights.
-  optional bool force_stable_sort = 6;
```

Comment:
I think this PR makes more changes than simply adding a new proto message. It makes me think that we don't really need a new proto message.

We can just have a `deterministic_order` bool flag in the `Sample` message. If it's false, then it's a normal sample. If it's true, we try to sort first, if can't sort, then cache the dataframe.

Reply:
ok.  @beliefer  let's follow @cloud-fan 's suggestion

Reply:
OK

Reply:
Please see https://github.com/apache/spark/pull/39240